cmd/libsnap-confine-private/cgroup-support.c

```diff
@@ -66,3 +67,18 @@ void sc_cgroup_create_and_join(const char *parent, const char *name, pid_t pid)
     }
     debug("moved process %ld to cgroup hierarchy %s/%s", (long)pid, parent, name);
 }
+
+static const char *cgroup_dir = "/sys/fs/cgroup";
+// from statfs(2)
+static const int CGROUP2_SUPER_MAGIC =  0x63677270;
+
+bool sc_cgroup_is_v2() {
+    struct statfs buf;
+
+    int err = statfs(cgroup_dir, &buf);
+    if (err == 0 && buf.f_type == CGROUP2_SUPER_MAGIC) {
+       fprintf(stderr, "WARNING: cgroup v2 is not fully supported yet\n");
```

Comment:
Nitpick: given that we call `sc_cgroup_is_v2` several times, could you please keep a `static int did_warn` flag to only print this once?